engines/order_management/app/services/order_management/reports/enterprise_fee_summary/summarizer.rb

```diff
@@ -0,0 +1,80 @@
+module OrderManagement
+  module Reports
+    module EnterpriseFeeSummary
+      class Summarizer
+        attr_reader :data
+
+        delegate :fee_type, :enterprise_name, :fee_name, :fee_placement,
+                 :fee_calculated_on_transfer_through_name, :tax_category_name, to: :representation
+
+        def initialize(data)
+          @data = data
+        end
+
+        def customer_name
+          data["customer_name"]
+        end
+
+        def total_amount
+          data["total_amount"]
+        end
+
+        def i18n_translate(translation_key, options = {})
+          I18n.t("order_management.reports.enterprise_fee_summary.#{translation_key}", options)
+        end
+
+        private
+
+        def representation
+          @representation ||= representation_klass.new(self)
```

Comment:
If we passed `data` instead of `self` we could avoid lots of references to `context` in all `DataRepresentations` classes.